2017 07 fr/idempiere

freddyheredia4 opened the pull request:
Mejoras en Terminos de pago, fechas y descripcion

freddyheredia4 committed:
Merged branch 10.0 into 2017-07FR/idempiere

freddyheredia4 committed:
Mejoras en manejo de fechas, terminos de pago, Referencia de Tercero

pepetreshere commented on connector_idempiere_salesorder/models/account_payment_term_custom.py:
```diff
@@ -0,0 +1,16 @@
+
+        # Columns
+    C_PaymentTerm_ID = fields.Integer('ID Payment Term from iDempiere',
+                              required=True,
```
Poner required = False, No deben ser obligatorio para todo el sistema Odoo, sino que debe validarse su configuración al momento de enviar los datos a idempiere, en ese momento debe lanzarse un raise (nótese que queremos que el usuario vea los errores que ocurren para que pueda corregirlos)

pepetreshere commented on connector_idempiere_salesorder/models/account_tax_custom.py:
```diff
@@ -9,4 +9,5 @@ class AccountTax(models.Model):
 
     # Columns
     C_Tax_ID = fields.Integer('ID tax from Idempiere',
+                              required=True,
```
Nuevamente, por las mismas razones, required=False

pepetreshere commented on connector_idempiere_salesorder/models/sales_order_custom.py:
```diff
@@ -15,6 +15,7 @@
 from idempierewsc.enums import DocAction
 from idempierewsc.request import SetDocActionRequest
 from datetime import datetime, date, time, timedelta
+from dateutil import parser
```
esta importación parece no ser utilizada, recomiendo removerla

pepetreshere commented on connector_idempiere_salesorder/models/sales_order_custom.py:
```diff
@@ -147,21 +153,20 @@ def sendOrder(self,connection_parameter,clienteid,order,product_setting,sales_or
-                                Field('PriceEntered', line.price_unit),
-                                Field('PriceActual', line.price_unit),
+                                Field('PriceEntered', (line.price_unit-(line.price_unit*line.discount)/100)),
+                                Field('PriceActual', (line.price_unit-(line.price_unit*line.discount)/100)),
```
Aqui me llama la atención como estas llenando estos campos, voy a revisar y te mando un email

freddyheredia4 committed:
Refactorizacion para consultar y crear registros en idempiere

En el modulo conector_idempiere se agregan metodos para facilitar la consulta de recordID y crear registros nuevos en odoo,, estos metodos se usan para validaciones al momento de crear terceros (Clientes) y evitar duplicados

freddyheredia4 committed:
Correciones Terceros, WebService CSV, Envio de pedidos
